src/main/java/minicraft/screen/entry/TextAreaEntry.java

```diff
@@ -0,0 +1,255 @@
+	}
+	public TextAreaEntry(String prompt, String regex, int maxLen, int maxLn, int lineSp, String initValue) {
+		super(prompt, regex, maxLen, initValue);
+		this.prompt = prompt;
+		this.regex = regex;
+		this.maxLength = maxLen > Short.MAX_VALUE ? Short.MAX_VALUE : maxLen;
+		this.maxLine = maxLn;
+		this.lineSpacing = lineSp;
+		
+		userInput = new ArrayList<>(Arrays.asList(initValue.split("\n")));
+	}
+	
+	@Override
+	public void tick(InputHandler input) {
+		int maxLen = maxLength == 0 ? Short.MAX_VALUE : maxLength;
+		if (input.getKey("cursor-down").clicked) {
+			int cursorX = cursor % Short.MAX_VALUE;
+			int cursorY = cursor / Short.MAX_VALUE;
+			String curString = userInput.get(cursorY);
+			if (cursorX+maxLen > curString.length()) {
+				if (cursorY+1 == userInput.size()) {
+					cursor = curString.length() + cursorY*Short.MAX_VALUE;
+					return;
+				}
+				String nextString = userInput.get(cursorY+1);
+				if (cursorX > nextString.length()) {
+					cursor = nextString.length() + (cursorY+1)*Short.MAX_VALUE;
+				} else {
+					cursor += Short.MAX_VALUE;
+				}
+			} else cursor = cursorX + maxLen+cursorY*Short.MAX_VALUE;
+			return;
+		} else if (input.getKey("cursor-up").clicked) {
+			int cursorX = cursor % Short.MAX_VALUE;
+			int cursorY = cursor / Short.MAX_VALUE;
+			if (cursorX-maxLen < 0) {
+				if (cursorY == 0) {
+					cursor = 0;
+					return;
+				}
+				String nextString = userInput.get(cursorY - 1);
+				if (cursorX > nextString.length()) {
+					cursor = nextString.length() + (cursorY-1)*Short.MAX_VALUE;
+				} else {
+					cursor -= Short.MAX_VALUE;
+				}
+			} else cursor = cursorX - maxLen+cursorY*Short.MAX_VALUE;
+			return;
+		} else if (input.getKey("cursor-left").clicked) {
+			int cursorX = cursor % Short.MAX_VALUE;
+			int cursorY = cursor / Short.MAX_VALUE;
+			if (cursorX == 0) {
+				if (cursorY == 0) return;
+				String nextString = userInput.get(cursorY - 1);
+				cursor = nextString.length() + (cursorY - 1)*Short.MAX_VALUE;
+			} else cursor--;
+			return;
+		} else if (input.getKey("cursor-right").clicked) {
+			int cursorX = cursor % Short.MAX_VALUE;
+			int cursorY = cursor / Short.MAX_VALUE;
+			String curString = userInput.get(cursorY);
+			if (cursorX == curString.length()) {
+				if (cursorY+1 == userInput.size()) return;
+				cursor = 0+(cursorY+1) * Short.MAX_VALUE;
+			} else cursor++;
+			return;
+		} else if (input.getKey("end").clicked) {
+			int cursorY = cursor / Short.MAX_VALUE;
+			cursor = userInput.get(cursorY).length() + cursorY*Short.MAX_VALUE;
+			return;
+		} else if (input.getKey("home").clicked) {
+			int cursorY = cursor / Short.MAX_VALUE;
+			cursor = 0+cursorY * Short.MAX_VALUE;
+			return;
+		} else if (input.getKey("page-up").clicked) {
+			if (pageUpListener!=null) pageUpListener.onChange(null);
+			return;
+		} else if (input.getKey("page-down").clicked) {
+			if (pageDownListener!=null) pageDownListener.onChange(null);
+			return;
+		}
```

Comment:
This code is awfully compact; please add more newlines. Replace the else if cluster with a switch statement.

Reply:
How can I use switch here? .clicked is boolean.